server/sonar-server/src/main/java/org/sonar/server/projectlink/ws/SearchAction.java

```diff
@@ -124,8 +124,8 @@ private ComponentDto getComponentByUuidOrKey(DbSession dbSession, SearchWsReques
       request.getProjectKey(),
       ComponentFinder.ParamNames.PROJECT_ID_AND_KEY);
 
-    if (!userSession.hasComponentUuidPermission(UserRole.ADMIN, component.uuid()) &&
-      !userSession.hasComponentUuidPermission(UserRole.USER, component.uuid())) {
+    if (!userSession.hasComponentPermission(UserRole.ADMIN, component) &&
```

Comment:
having a method `checkComponentAnyPermission(ComponentDto component, String permission, String... otherPermissions)` would allow removing the `throw insufficientPrivilegesException()` from the action's code base, couldn't it?

Removing the need to write custom code like this from action is a step in the direction of more consistent handling of errors (consistent messages, consistent HTTP error codes, ...).

Although I favorite having the `ComponentDto` as first argument of all the `*ComponentPermission` method for personal taste, the method suggested above gives a non subjective argument in this direction.